### ISSUE TITLE
All shares are shown as Unmounted after boot and Samba not properly configured

### ISSUE THREAD

**Issue by stefanogreg** [rockstor/rockstor-core#2535]
I'm using ROCKSTOR 4.5.8-0 and since last update at boot time a common issue already reported in the community forum shows up:
all shares shows as unmounted

Following the instructions in the forum I've made a small script to enable quotas on all shares and restart rockstor and rockstor-bootsrap
i run this script with a crontab @reboot and it seems to work,

During my initial investigation I also had issue with configuring smb shares and I noticed an error in the logs showing that the mnt-share script was not found
then looking into /opt/rockstor i found that a .venv was added to isolate rockstor Python environment from the system one which is really good

However this broke Samba module (and possibly others) that seems to not be aware of this change for the pre-existing shares and left them (during update) in such form

```
[MyShare]
root preexec = "/opt/rockstor/bin/mnt-share MyShare"
```

To fix the issue I manually edited (even if discouraged) the smb.conf file to look for bin folder inside the .venv and this worked really well

```
[MyShare]
root preexec = "/opt/rockstor/.venv/bin/mnt-share MyShare"
```

I do suspect the addition of the .venv could be the cause of the improper mounting of the shares at boot, may be the mnt-share is still invoked on the old path

as a reference the my fix-shares.sh script that makes the shares properly mounted:
```
#!/bin/sh                                                                                                                              
echo "Enabling quotas"                                                                                                                 
btrfs quota enable /mnt2/ROCKS_ON/                                                                                                     
btrfs quota enable /mnt2/EVO870_DATA                                                                                                   
btrfs quota enable /mnt2/LOCAL                                                                                                         
                                                                                              
echo "Restarting rockstor"                                                                                                             
systemctl restart rockstor                                                                                                             
systemctl restart rockstor-bootstrap 
```
and my entry in crontab:
```
stefano@rocky:~> sudo crontab -l                                                                                                       
@reboot /home/stefano/fix-shares.sh 
```

**Comment by phillxnet** [comment]
@stefanogreg Excellent report, and my apologies for not already having created an issue for this. @FroggyFlox and I have discussed the creation of said issue (now in existence here thanks to yourself) but unfortunately we are both pulled in multiple directions currently.

You are spot on regarding our .venv change which came about as a result of tending to a major aspect of our current technical dept by way of us moving (updating) to a set version of Poetry. See:
https://github.com/rockstor/rockstor-core/pull/2434

The plan to address what you saw (and fixed for yourself) old paths in samba config, was an add some more migration mechanisms to initrock that simply look to see if legacy paths exist and switch them out if they are found.

I've added this, as it's planned counterpart was bound, to our next stable milestone so tend to those folks running our more production orientated Stable rpm releases. Unfortunately time has gotten away with us and it made more sense to start folks out on our Release Candidates via our installers.

Forum post relating to this find:
https://forum.rockstor.com/t/smb-shares-inaccesible-after-v4-5-4-0-upgrade/8627
Where @FroggyFlox diagnosed the exact same issue re SMB.
Again my apologies for dropping the ball and not creating an appropriate issue back then.

We have had a number of these knock-ons, inevitably give the massive change, but as you say we are far better off now we have Poetry and keeping the .venv explicit in our paths helps with diagnosing future issues rather than just doing a patch via a sym-link or the like. But is has brought some breakage along the way. We I think also have some  'ensuring'/migration to do on our scheduled tasks.

I'll add this issue to our current Milestone.

Thanks again for the creation/contribution. And for sharing your work arounds. We are getting there now post the build system change and our next major adventure will involve moving (finally) to a more modern version of Python.
